app/src/main/kotlin/com/wire/android/workmanager/worker/SingleUserMigrationWorker.kt

```diff
@@ -0,0 +1,116 @@
+package com.wire.android.workmanager.worker
+
+import android.content.Context
+import androidx.core.app.NotificationCompat
+import androidx.hilt.work.HiltWorker
+import androidx.lifecycle.asFlow
+import androidx.work.BackoffPolicy
+import androidx.work.Constraints
+import androidx.work.CoroutineWorker
+import androidx.work.ExistingWorkPolicy
+import androidx.work.ForegroundInfo
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.OutOfQuotaPolicy
+import androidx.work.WorkInfo
+import androidx.work.WorkManager
+import androidx.work.WorkRequest.MIN_BACKOFF_MILLIS
+import androidx.work.WorkerParameters
+import androidx.work.await
+import androidx.work.workDataOf
+import com.wire.android.R
+import com.wire.android.migration.MigrationData
+import com.wire.android.migration.MigrationManager
+import com.wire.android.migration.getMigrationProgress
+import com.wire.android.migration.toData
+import com.wire.android.notification.NotificationChannelsManager
+import com.wire.android.notification.NotificationConstants
+import com.wire.android.notification.openAppPendingIntent
+import com.wire.kalium.logic.data.id.QualifiedIdMapperImpl
+import com.wire.kalium.logic.data.user.UserId
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.map
+import java.util.concurrent.TimeUnit
+
+@HiltWorker
+class SingleUserMigrationWorker @AssistedInject constructor(
+    @Assisted appContext: Context,
+    @Assisted workerParams: WorkerParameters,
+    private val migrationManager: MigrationManager,
+    private val notificationChannelsManager: NotificationChannelsManager
+) : CoroutineWorker(appContext, workerParams) {
+
+    override suspend fun doWork(): Result = coroutineScope {
+        val userId = inputData.getString(USER_ID_INPUT_DATA)?.let { userId ->
+            QualifiedIdMapperImpl(null).fromStringToQualifiedID(userId)
+        } ?: return@coroutineScope Result.failure()
+
+        when (migrationManager.migrateSingleUser(userId, this) { setProgress(it.type.toData()) }) {
+            is MigrationData.Result.Success -> Result.success()
+            is MigrationData.Result.Failure -> Result.retry()
```

Comment:
Do we want to retry the process automatically if it failed? I think it won't change anything until we update the app, right? I'm implementing this error screen that tells the user that unfortunately not all messages could be migrated so we should probably just fail here and show it to the user then 🤔